cocos/ui/UIRichText.cpp

```diff
@@ -1609,110 +1646,94 @@ void RichText::handleTextRenderer(const std::string& text, const std::string& fo
+                pushToContainer(textRenderer);
+                break;
             }
-            if (leftRenderer)
+
+            int leftLength = 0;
+            if (wrapMode == WRAP_PER_WORD)
+                leftLength = findSplitPositionForWord(textRenderer, utf8Text, _leftSpaceWidth, _customSize.width);
+            else
+                leftLength = findSplitPositionForChar(textRenderer, utf8Text, _leftSpaceWidth, _customSize.width);
+
+            // split string
+            if (leftLength > 0)
             {
-                leftRenderer->setColor(color);
-                leftRenderer->setOpacity(opacity);
-                pushToContainer(leftRenderer);
-
-                if (flags & RichElementText::ITALICS_FLAG)
-                    leftRenderer->enableItalics();
-                if (flags & RichElementText::BOLD_FLAG)
-                    leftRenderer->enableBold();
-                if (flags & RichElementText::UNDERLINE_FLAG)
-                    leftRenderer->enableUnderline();
-                if (flags & RichElementText::STRIKETHROUGH_FLAG)
-                    leftRenderer->enableStrikethrough();
-                if (flags & RichElementText::URL_FLAG)
-                    leftRenderer->addComponent(ListenerComponent::create(leftRenderer,
-                                                                         url,
-                                                                         std::bind(&RichText::openUrl, this, std::placeholders::_1)));
-                if (flags & RichElementText::OUTLINE_FLAG) {
-                    leftRenderer->enableOutline(Color4B(outlineColor), outlineSize);
-                }
-                if (flags & RichElementText::SHADOW_FLAG) {
-                    leftRenderer->enableShadow(Color4B(shadowColor), shadowOffset, shadowBlurRadius);
-                }
-                if (flags & RichElementText::GLOW_FLAG) {
-                    leftRenderer->enableGlow(Color4B(glowColor));
-                }
+                textRenderer->setString(utf8Text.getAsCharSequence(0, leftLength));
+                pushToContainer(textRenderer);
             }
-        }
 
-        addNewLine();
-        handleTextRenderer(cutWords, fontName, fontSize, color, opacity, flags, url,
-                           outlineColor, outlineSize,
-                           shadowColor, shadowOffset, shadowBlurRadius,
-                           glowColor);
-    }
-    else
-    {
-        textRenderer->setColor(color);
-        textRenderer->setOpacity(opacity);
-        pushToContainer(textRenderer);
+            // skip spaces
+            StringUtils::StringUTF8::CharUTF8Store& str = utf8Text.getString();
+            int rightStart = leftLength;
+            while (rightStart < (int)str.size() && str[rightStart].isASCII() && std::isspace(str[rightStart]._char[0], std::locale()))
```

Comment:
why need to skip spaces?

Reply:
In original code, it skips one space.
But I think skipping all spaces may be better.

eg(_ instead of space): "abcd________________123"
suppose it is splitted after letter 'd'
Should it desplay like

abcd
123

or

abcd
________________123
? please give me some advice

Reply:
I have not idea. How it been displayed on Android/iOS UI element? I think can have the same behavior as them.

Reply:
I test it with iOS's webview. It trims the spaces. see below

Reply:
K, thanks.